### PR TITLE
[DOCS] Adding introductory documentation to the launch buttons

### PR DESCRIPTION
**Pull request by choldgraf**
## References

In preparation for adding a retrolab tutorial to the top row of try.jupyter.org, I thought it'd be helpful to add two quick introductory notebooks for the Lab / Retrolab buttons on the JupyterLite docs. This way, users see some content when they click the button, and can have a few pointers in the right direction. This may be a short-term solution until we can flesh out some more introductory content that is fitting for try.jupyter.org, but what do people think about this as a first step?

- https://github.com/jupyter/jupyter.github.io/issues/513
- https://github.com/jupyterhub/team-compass/issues/463

## Documentation changes

- There are now two lightweight "introductory" notebooks that are meant to guide people on their first steps with the interfaces.
- The buttons in the top left of the docs now open an `intro` notebook when JupyterLite is loaded.

### CODE REVIEW

**Comment by bollwyvl** [comment]
This is good content.

Given the size breakdown on that other issue, I don't think we'd necessarily want to _always_ point directly at a particular python notebook with its 30mb load as the very first introduction. If it was a notebook, a JS kernel notebook would be _far_ lighter weight, and is more likely to stay in this repo, and could still make some pretties with e.g. vega. However, more likely, we'd want to rework the prose into steps of a tour, and start that off.

Over on https://ipydrawio.readthedocs.io and https://pidgy.readthedocs.io i've played around with some other approaches to flexibly launching the demo with some URL-based configuring... perhaps we consider some alternate forms, including offering all of the examples as potential starting points.

More broadly: even if we _did_ bring the capability into `try.jupyter.org`, this project's `main` is not ready to be the destination, even if the released package could build a site, deployed on `try.jupyter.org/lite/`, for example, that was. We do what we can to keep it working here, but you know...

**Comment by choldgraf** [comment]
Gotcha - makes sense to me. Do you think we should close this PR? Or rework the content somehow? I suspect that many people will land on the jupyterlite docs and just click one of the try buttons in the top left, since that's a high eyeballs spot, and might be confused at the lack of context.

I didn't realize size was a concern (WASM and jupyterlite continues to work in mysterious ways to me haha). Maybe it should just be a markdown file?

**Comment by jtpio** [comment]
Thanks @choldgraf!

> Do you think we should close this PR? Or rework the content somehow? 

Probably we can still keep the `intro.ipynb` and `intro-retro.ipynb` as they would show up in the file browser at the top-level.

Another idea would be to open `examples/README.md` by default with links to these notebooks. And iterate a bit on the `README.md` to make it more like a proper gallery of examples (could be done at a later stage).

File: examples/README.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
 # JupyterLite Examples
 
 This folder contains example notebooks and files demoing what is possible to do with
-JupyterLite.
+JupyterLite. To get started, there is a short introduction in [the `intro.ipynb` notebook](intro.ipynb).
```
Comment (jtpio) [inline examples/README.md:4]: Pushing a commit to fix the lint failure.

Comment (jtpio) [reply examples/README.md]: Done in https://github.com/jupyterlite/jupyterlite/pull/432/commits/6a054a618bbe59bccf89eb9964216e130e51dba0

**Comment by choldgraf** [comment]
> Another idea would be to open examples/README.md by default with links to these notebooks. And iterate a bit on the README.md to make it more like a proper gallery of examples (could be done at a later stage).

This sounds like a nice idea - let me know if you'd like this in the current PR or not.

**Comment by jtpio** [comment]
@choldgraf if you would like to do it in this PR that would be great, thanks!

**Comment by choldgraf** [comment]
ok - can you give me an idea of what kind of content you think is best there? Something like a few paragraphs about JupyterLite in general with pointers to the intro notebooks? Anything else?

**Comment by jtpio** [comment]
That would already be a good introduction yes.

There was also some previous discussions about this in https://github.com/jupyterlite/jupyterlite/issues/168 and related issues.

**Comment by jtpio** [comment]
Just rebased onto `main` to grab the latest updates.

Testing on RTD, opening a notebook by default doesn't feel too heavy at first. Although it's true some dependencies are pulled, and the first cell run still takes some time but it's a bit faster now thanks to https://github.com/jupyterlite/jupyterlite/pull/433.

**Comment by jtpio** [comment]
However there seems to be a `micropip` related issue when opening the notebook with the new Pyodide 0.19 and lite `0.1.0a19`:

image

But this does not seem to be a problem when testing on https://jupyterlite.github.io/demo/, which doesn't pre-cache the wheels:

image

**Comment by jtpio** [comment]
> However there seems to be a micropip related issue when opening the notebook with the new Pyodide 0.19 and lite 0.1.0a19:

Ah nevermind that, this was most likely a caching issue. Trying again in a private window seems to work fine.

File: docs/_templates/launch.html [hunk 2]
```diff
@@ -13,7 +13,7 @@
 </a>
 
 <a
-  href="{{ pathto('_static/retro/index.html', 1) }}"
+  href="{{ pathto('_static/retro/', 1) }}?path=notebooks/intro-retro.ipynb"
```
Comment (jtpio) [inline docs/_templates/launch.html:16]: Looks like removing the `index.html` here creates an infinite redirect loop?

https://user-images.githubusercontent.com/591645/149344871-68ecb47f-2b17-4d43-a8e0-fa2a6e367966.mp4

Comment (bollwyvl) [reply docs/_templates/launch.html:16]: yeah, different hosts react differently with respect to `index.html`, trailing slashes, etc. and `index.html` is the only one that's going to _work_ work. That's why the rediraffe stuff in the first place for "pretty" urls.

**Comment by bollwyvl** [comment]
> Format intro notebooks

I've been wondering: we should probably add a no-op `nbconvert --to notebook` to our linter, as the "one line of JSON" is _even less_ ergonomic than a regular ipynb :rofl:

File: docs/_templates/launch.html (hunk 2 above)
Comment (jtpio) [reply docs/_templates/launch.html:16]:
```suggestion
  href="{{ pathto('_static/retro/index.html', 1) }}?path=notebooks/intro-retro.ipynb"
```

File: docs/_templates/launch.html [hunk 3]
```diff
@@ -13,7 +13,7 @@
 </a>
 
 <a
-  href="{{ pathto('_static/retro/index.html', 1) }}"
+  href="{{ pathto('_static/retro/index.html', 1) }}?path=notebooks/intro-retro.ipynb"
```
Comment (jtpio) [inline docs/_templates/launch.html:16]: The format is slightly different for retro:

```suggestion
  href="{{ pathto('_static/retro/notebooks/index.html', 1) }}?path=intro-retro.ipynb"
```

**Comment by jtpio** [comment]
Since this is good content, I would be in favor of getting this in.

We can keep the demo link in the project description of the repo (and similar links) as it is now without the `?path=`:

image

cc @bollwyvl since you had a comment about loading times above

**Comment by choldgraf** [comment]
@jtpio I can probably put a quick cycle into this (say, 30 minutes) if I have a really clear idea of what new content needs to be in here, but probably don't have time to do more complex iterations. Let me know if you'd prefer to just merge it in, or if I should make some changes first

**Comment by jtpio** [comment]
Thanks @choldgraf.

> Another idea would be to open examples/README.md by default with links to these notebooks. And iterate a bit on the README.md to make it more like a proper gallery of examples (could be done at a later stage).

Actually the `README` opens with the text editor by default for now and not the markdown preview, which makes opening that file with `?path=` less interesting I believe. Although we could override the preferred factory for markdown files to open with the markdown preview on the demo ReadTheDocs deployment.

Which is why we could just keep the current content as it is with this PR, and iterate later if needed.

**Review by jtpio — APPROVED** [review]
Thanks!

**Comment by choldgraf** [comment]
thanks for the iteration and suggestions all :-)

**Pull request merged**